### PR TITLE
code review

### PR DESCRIPTION
**Pull request by aolszewski**

### CODE REVIEW

File: js/nyt.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+var inputMonth = $("#input-month").val();
+            //user selected day (from dropdown menu)
```
Comment (aolszewski) [inline js/nyt.js:2]: your indentation should be consistent

File: js/script.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    ];
+
+    //when the user submits their information...
+    $("#submitBtn").on("click", function (){
+        
+        $(".card-title-area").empty();
+        //get the users input name
+        var inputName = $("#input-name").val();
+        //if the user doesn't enter a name, return
+        if (inputName === "") {
```
Comment (aolszewski) [inline js/script.js:54]: FYI, you can probably shorten this to a falsey check 

```if (!inputName) {```

it's a little more concise and it also deals with false or null

File: js/script.js [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+            return;
+        }else {
```
Comment (aolszewski) [inline js/script.js:56]: your spacing should be consistent, it makes it easier to read

File: js/script.js [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+
+    function makeCard() {
+        //clear out current text
+        $(".card-text-area").empty();
+
+        //user's selected card style is selected by value of book, space, or paper
+        var cardStyle = $("#card-styles").val();
+        console.log(cardStyle);
+
+        if (cardStyle === null) {
```
Comment (aolszewski) [inline js/script.js:83]: same here for the falsey check, this can be

```if (!cardStyle)```

File: js/script.js [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+        var inputYear = $("#input-year").val();
+
+        //make a variable to hold our card image or headlines
+        var cardImage = $("<img>");
+        
+        //make api keys and get functions for each api 
+        //nasa API Function
+        function getNasaImage () {
+            //make query url
+            var nasaURL = "https://api.nasa.gov/planetary/apod?api_key=PxXYhbxjF4qoZihVRPPnN3C4IpvxKF4N1eWiJVRx&date=" +
```
Comment (aolszewski) [inline js/script.js:145]: this is a very minor thing, but it's a best practice to put the `+` at the beginning of the line

it makes it easier to read and less error prone, you would be surprised how often you end up missing the plus at the end

File: js/script.js [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+                //add an alt
+                cardImage.attr("alt", "book cover image");
+                //apend it to displayed card
+                $(".api-image").append(cardImage);
+            })
+        }
+        //TMDB API Function
+        function getMovieImage() {
+            //make query URL
+            var movieURL = "https://api.themoviedb.org/3/search/movie?api_key=041542fa1899303e8c187b3a599c9d43&language=en-US&query=e&include_adult=false&primary_release_year=" +
```
Comment (aolszewski) [inline js/script.js:196]: you want to break up the url a little bit since this line is so long, having to scroll to the right to read is a pain

File: js/script.js [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+                url: movieURL,
+            }).then(function(response){
+                //find the movie poster in the API and then use that to make the full image url
+                var poster = null;
+                //until there is a non-null value for a movie poster, the api will continue to look
+                for (i=0; i<response.results.length && !poster; i++) {
+                    var current = response.results[i];
+                    if (current.poster_path) {
+                        poster=current.poster_path;
+                        var moviePosterFull = "https://image.tmdb.org/t/p/original" + poster;
```
Comment (aolszewski) [inline js/script.js:212]: you usually want to declare variables before you enter blocks... this is fine because it's a var, but when we start using `let` and `const` this would not have worked

try to declare variables in the same block they are used, so

```var poster = null;
var moviePosterFull = null;

...
for (...)
     if (current.poster_path) {
          moviePosterFull = "https://image.tmdb.org/t/p/original" + poster;;
...

File: js/script.js [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+            })            
+        }
+    
+        //if the user selects "space", run the getNasaImage function
+        if (cardStyle === "space") {
+            getNasaImage();
+        //if the user selects "book", run the getBookImage function
+        } else if (cardStyle === "book") {
+            getBookImage();
+        // } else if (cardStyle === "paper") {
```
Comment (aolszewski) [inline js/script.js:230]: you should remove commented out code since it makes it harder to read the code that is actually being executed

File: index.html [hunk 9]
```diff
@@ -0,0 +1,167 @@
+<!DOCTYPE html>
```
Comment (aolszewski) [inline index.html:1]: html looks really good

File: js/script.js [hunk 10]
```diff
@@ -0,0 +1,253 @@
+//wait until the page is ready to load
+$( document ).ready(function() {
+
+    //populate the day option drop-down with the days of the month up to 31
+    //for each possible day of the month (31)
+    for (var i=0; i<31; i++) {
```
Comment (aolszewski) [inline js/script.js:6]: you should probably extract this to a function

**Review by aolszewski — COMMENTED** [review]
Great job, most of my comments are best practice or code style concerns.

Looks great and works great!

File: index.html [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+<html lang="en">
+    <head>
+        <meta charset="utf-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+        <title>Birthday Card Generator</title>
+        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.0/css/bulma.min.css">
+        <link href="https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,300;0,400;0,600;1,200&display=swap" rel="stylesheet">
```
Comment (erellsworth) [inline index.html:8]: Just FYI, Google Fonts does allow you to embed multiple different fonts with one single link tag.

File: index.html [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+                            <div class="columns">
+                                <div class="column is-half is-offset-one-quarter choose-bday">
+                                    <div class="content">
+                                        <h1>Choose Birthday:</h1>
+                                    </div>
+                                </div>
+                            </div>
+                            <div class="columns">
+                                <div class="column is-2 is-offset-one-quarter">
+                                    <div class="select is-info" style="width: 100%">
```
Comment (erellsworth) [inline index.html:51]: It's a best practice to avoid in-line styles if at all possible. CSS should be kept to CSS files or style tags.

File: js/nyt.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+            var inputYear = $("#input-year").val();
+
+function getHeadlines() {
+            var headlineArea = $("<div>");
+            var headlines = $("<h5>");
+            var links = $("<a>");
+
+            var headlinequeryURL = "https://api.nytimes.com/svc/search/v2/articlesearch.json?begin_date="
+                + inputYear + inputMonth + inputDay + "&end_date="
+                + inputYear + inputMonth + inputDay + "&fq=print_page:1&print_section:A&api-key=l150AJIrBZQqHoU50PKwaI3cMa29A9da";
```
Comment (erellsworth) [inline js/nyt.js:14]: I highly recommend putting API keys into variables.
